### PR TITLE
Remove unneeded scc

### PR DESCRIPTION
**Pull request by adrianchiris**

### CODE REVIEW

File: deploy/role.yaml [hunk 1]
```diff
@@ -252,9 +294,113 @@ rules:
   - update
   - watch
 - apiGroups:
-  - ""
+  - mellanox.com
```
Comment (ykulazhenkov) [inline deploy/role.yaml:297]: in the ClusterRole we have access to `mellanox.com` apiGroup  with `'*'` resources. Do we really need to add each resource explicitly here?

Comment (adrianchiris) [reply deploy/role.yaml:297]: Thats auto-generated code, i did not edit this explicitly.

**Review by rollandf — APPROVED** [review]
LGTM

File: deploy/role.yaml (hunk 1 above)
Comment (ykulazhenkov) [reply deploy/role.yaml:297]: It looks like at some point we manually added the wildcard rule. Ack, let's keep specific rules and wildcard as a fallback.

**Review by ykulazhenkov — APPROVED** [review]

File: deploy/role.yaml (hunk 1 above)
Comment (adrianchiris) [reply deploy/role.yaml:297]: hmm lemme check this further.

Comment (adrianchiris) [reply deploy/role.yaml:297]: we got wildcards set in the following reconcilers:

https://github.com/Mellanox/network-operator/blob/84f986f33d0753e8dacfe946bca05bd994f85fe9/controllers/ipoibnetwork_controller.go#L53
https://github.com/Mellanox/network-operator/blob/84f986f33d0753e8dacfe946bca05bd994f85fe9/controllers/macvlannetwork_controller.go#L55
https://github.com/Mellanox/network-operator/blob/84f986f33d0753e8dacfe946bca05bd994f85fe9/controllers/nicclusterpolicy_controller.go#L58

hostdevice controller seems to be the only saint who doesnt use wildecards :)  

ill create an issue to remove the wildecards. it should be handled in a separate PR

**Pull request merged**